airbyte-cdk/python/airbyte_cdk/sources/declarative/partition_routers/substream_partition_router.py

```diff
@@ -25,14 +27,14 @@ class ParentStreamConfig:
     request_option: How to inject the slice value on an outgoing HTTP request
     """
 
-    stream: Stream
+    stream: "DeclarativeStream"
```

Comment:
there's a circular dependency because we import from the __init__ files. [The type is imported, but only for type checking](https://github.com/airbytehq/airbyte/pull/35471/files#diff-423f9b3e3269c4750965aae83f8df00ffa4eeb0098f91aa0c56a150c98b328d6R15-R16)